### DISCUSSION TITLE
Possible modification for better code

### DISCUSSION THREAD

**Discussion by lakbychance** [0xTheProDev/Operational-Transformation#97] [General]
https://github.com/Progyan1997/Operational-Transformation/blob/c743cc51648d544c1a783b66e0c81dc208e0e593/packages/ace/src/tooltip-marker.ts#L52

Hey @Progyan1997 , I thought about this code here and I feel that this piece of code works for now but depends on `querySelector` to retrieve the `ace_content` div. Instead of that I think we can shift this code inside the `update` method and use `closest` to append the tooltip marker like so :- 

```
    if (!this._aceContent) {
      this._aceContent = markerLayer.element.closest('.ace_content');
      this._aceContent.append(this._tooltipWidget);
    }
```

where `_aceContent` is a new property on `TooltipMarker` class. We can also not introduce this new property and instead run the `closest` and `append` operation every time which will result in same behavior. 

What do you think ? If you would wanna go with this, would you choose introducing `_aceContent` or performing `closest` and `append` every time ?

My intuition for this change - The reason is possible collision of same class names can break this with top to bottom existing approach. In bottom to top with `closest` the probability is very less unless user manipulates the editor DOM nodes itself.

**Comment by 0xTheProDev** [comment]
Agreed. We should short-circuit the lookup and can have this around conditional to avoid unnecessarily DOM access.
